src/main/java/interfaces/IRequestBuilder.java

```diff
@@ -1,5 +1,6 @@
 package com.td.HttpServer;
+import java.io.IOException;
 
 public interface IRequestBuilder {
-  public HttpRequest createRequest(String rawRequest) throws InvalidHttpRequestException;
+  public HttpRequest createRequest(String rawRequest) throws IOException, InvalidHttpRequestException;
```

Comment:
Why are we all of a sudden exposing the low level `IOException`?

Two things about this:
1. Ideally, anything using this interface only has to worry about exceptions that are at the same level as this interface. `IOException` is certainly below that.
2. It's very suspicious that an _interface_ has to change to all of a sudden declare that it may throw this exception. Seems like the interface is getting coupled to the thing that implements this interface, which should _never_ be the case. If an interface changes, it's because the client of the interface requires it to change.
3. Fill in the blank: "Point #2 above is the _ in SOLID"